src/ServicePulse.Host/app/js/views/message/controller.js

```diff
@@ -0,0 +1,99 @@
+﻿;
+(function(window, angular, undefined) {
+    "use strict";
+
+    function controller(
+        $scope,
+        $routeParams,
+        scConfig,
+        toastService,
+        serviceControlService,
+        archivedMessageService) {
+
+        var vm = this;
+        
+        vm.message = {};
+        
+        var init = function () {
+            var messageId = $routeParams.messageId;
+            vm.loadMessage(messageId);
+        }
+
+        vm.clipComplete = function(messageId) {
+            toastService.showInfo(messageId + ' copied to clipboard');
+        };
+
+        vm.togglePanel = function (message, panelnum) {
+            if (!angular.isDefined(message.messageBody)) {
+                serviceControlService.getMessageBody(message.message_id).then(function (msg) {
+                    message.messageBody = msg.data;
+                }, function () {
+                    message.bodyUnavailable = "message body unavailable";
+                });
+            }
+
+            if (!angular.isDefined(message.messageHeaders)) {
+                serviceControlService.getMessageHeaders(message.message_id).then(function (msg) {
+                    message.messageHeaders = msg.data[0].headers;
+                }, function () {
+                    message.headersUnavailable = "message headers unavailable";
+                });
+            }
+            message.panel = panelnum;
+            return false;
+        };
+
+        vm.retryMessage = function () {
+            serviceControlService.retryFailedMessages([vm.message.message_id]);
```

Comment:
Can we change the `retryFailedMessages` method to return the promise so that it's the same as the `restoreMessageFromArchive` method?

It also means we can chain the `vm.message.retried = true;` line onto only the success portion of the POST. In case the network is down or something goes wrong with the actual HTTP comms, then we won't set the retried flag to true incorrectly.